vcl_templates/www.vcl.erb

```diff
@@ -327,6 +323,12 @@ sub vcl_fetch {
       set beresp.ttl = 900s;
       set beresp.http.Cache-Control = "max-age=900";
     }
+
+    # Only cache non-GET/HEAD requests if they return a 404 or 405
+    if (!(req.request == "GET" || req.request == "HEAD") && !http_status_matches(beresp.status, "404,405")) {
+      set beresp.ttl = 0s;
+      set beresp.cacheable = false;
+    }
```

Comment:
The intent of this change is to avoid cacheing non-GET/HEAD requests if their response status is not 404,405.

One thing I am not clear on is whether the 405 response will be cached. By the Varnish default 405 responses are not cached. So I don't know if we need something like this in there:
```
    if (http_status_matches(beresp.status, "405")) {
      # cache the 405 response.
    }
```

Reply:
> ### beresp.cacheable
> True if the request resulted in a cacheable response. A response is considered cacheable if HTTP status code is 200, 203, 300, 301, 302, 404 or 410 and pass wasn’t called in vcl_recv. If however, both the TTL and the grace time for the response are 0 beresp.cacheable will be 0.
> beresp.cacheable is writable.

So it sounds like adding the above with `beresp.cacheable = true` would be necessary, as setting the default ttl may not be enough.

I feel like I need to read the source code of Varnish (and Fastly's implementation which is _based on_ Varnish 2.1) to know for sure what is necessary.